File: NumberGuessingGame.cs
using System;

namespace number.guessing.game {
    class Program {
        static void Main(string[] args) {
            Random random = new Random();
            int randomNum = random.Next(1, 100);
            int guess;
            int guessCount = 0;
            while (true) {
                Console.Write("Guess a number between 1-100: ");
                guess = Convert.ToInt32(Console.ReadLine());
                guessCount++;
                if (guess == randomNum) {
                    Console.WriteLine("You won!");
                    Console.WriteLine("It took you " + guessCount + " guesses");
                    Console.Write("Press Any Key To Continue . . .");
                    Console.ReadLine();
                    break;
                }
                else if (guess > randomNum) {
                    Console.WriteLine("Your guess was too high, try again!");
                }
                else if (guess < randomNum) {
                    Console.WriteLine("Your guess was too low, try again!");
                }
            }
        }
    }
}

File: NumberGuessingGame.rs
use rand::Rng;
use std::cmp::Ordering;
use std::io;

fn main() {
    let secret_number = rand::thread_rng().gen_range(1, 101);
    loop {

        println!("Guess a number between 1 - 100!");
        let mut guess = String::new();

        io::stdin()
            .read_line(&mut guess)
            .expect("Failed to read line");

        let guess: u32 = match guess.trim().parse() {
            Ok(num) => num,
            Err(_) => continue,
        };

        match guess.cmp(&secret_number) {
            Ordering::Less => println!("Your guess was too small, try again."),
            Ordering::Greater => println!("Your guess was too big, try again."),
            Ordering::Equal => {
                println!("You win!");
                break;
            }
        }
    }
}